### ISSUE TITLE
KNOWN ISSUE: some artnet packets are dropped when many universes are sent to the controller

### ISSUE THREAD

**Issue by cyberboy666** [cyberboy666/lateral_led_controller#2]
_tl;dr - if using ethernet connection then switching to 10mbps can improve performance - even so a maximum of 12 universes (~2000 leds) is recommended due to this current bottleneck - i do believe it is possible to atleast x2 this with firmware updates_

## details of the issue

in the artnet protocol up to 512 values can be sent in a single packet - this is called a __universe__ and since we need 3 values per led (rgb) then a single universe can control up to 170 leds. if we want to control more than this we need to split this information across multiple universes (packets)

when sending lots of universes (our installation needed 12) i was observing that packets were being dropped. on the controller side i printed out to serial average number of universes collected per frame so i could track this. 

the amount of dropped packets reduced significantly when i switched from 100mbps to 10mbps wired connection. 

its my assumption that since these multiple universes for the same frame are being sent within very quick succession that the esp32 buffer is overflowing / not able to process them all in time.

this [discussion on reddit](https://www.reddit.com/r/FastLED/comments/14mfelz/comment/jq1u1g3/) seems to support this assumption:

> It can be done. The limiting factor is 10 mb/s ethernet. If you use 100 mb/s you will lose some packets since Resolume sends them in bursts and the receive buffer will overflow. At 10 mb/s you can send 5400 pixels at 60 fps. 

(i could not reproduce 5400 pixels at 60 fps at 10mbps but i think they are using a [different library](https://github.com/hpwit/artnetesp32v2) underneath to us here) 

## firmware improvements ?

i spent some time looking into this but did not find a way to improve it further than switching to 10mbps connection.

for now im listing the specs on this controller to supporting up to `12 universes` = 12*170= __2040 leds__ which is the max i have confirmed it to work smoothly for.

theres also the other bottleneck of the protocol speed of WS281X which runs at 800kbps which allows for up to 240 leds per data line (while updating above 30fps) which results in a total of 240*8 = __1920 leds__

however it does sound like there are other optimizations available (overclocking ws281x slightly + less other overheads maybe?) that can allow even ws281x leds to output 540 pixels per line at 60fps (total of 4320 leds which is around 26 universes):

> It takes 16.2 ms plus begin and end pulses. That's more than the 16.6 milliseconds that one frame at 60 fps takes. You can either run your system at less than 60 fps or you can overdrive your leds at 833 kHz instead of the standard 800 kHz. At 833 kHz it takes 15.6 ms plus begin and end pulses to send 540 pixels. At that speed you can do 60 fps
